app/assets/javascripts/managers.js.coffee

```diff
@@ -0,0 +1,23 @@
+App.Managers =
+
+  generatePassword: ->
+    chars = 'aAbcdeEfghiJkmnpqrstuUvwxyz23456789$!'
+    pass = ''
+    x = 0
+    while x < 8
+      i = Math.floor(Math.random() * chars.length)
+      pass += chars.charAt(i)
+      x++
+    return pass
+
+  initialize: ->
+    $(".generate-random-value").on "click", (event) ->
+      password = App.Managers.generatePassword()
+      $('#user_password').val(password)
+
+    $(".show-password").on "click", (event) ->
+      console.log("hey")
+      if $("#user_password").is("input[type='password']")
+        $('#user_password').prop 'type', 'text'
+      else
+        $('#user_password').prop 'type', 'password'
```

Comment:
If an object is going to be used more than twice it's a good practice to store it into a local variable.